### PR TITLE
Fix react/no-access-state-in-setstate errors.

### PR DESCRIPTION
**Pull request by XhmikosR**
@Mcat12 does this look good to you?

### CODE REVIEW

**Comment by XhmikosR** [comment]
I also have https://github.com/pi-hole/web/compare/development...lint-3 in the works, I plan to make a PR later, but feel free to comment on the patches themselves :)

BTW can we tweak Codecov not to mark the build as failed?

File: package.json [hunk 1]
```diff
@@ -96,9 +96,12 @@
   },
   "eslintConfig": {
     "root": true,
-    "extends": ["react-app"],
+    "extends": [
+      "react-app"
+    ],
```
Comment (Mcat12) [inline package.json:101]: It turns out we don't need the array:
```suggestion
    "extends": "react-app",
```

**Comment by Mcat12** [comment]
Codecov is not set as required, and one of the two checks was successful (overall coverage didn't change). Considering the changes it can be ignored.

File: package.json (hunk 1 above)
Comment (XhmikosR) [reply package.json:101]: Yeah, but I think I'm gonna keep it because I plan to add more later.

**Comment by XhmikosR** [comment]
>Codecov is not set as required, and one of the two checks was successful (overall coverage didn't change). Considering the changes it can be ignored.

True, but it still marks the check as failed. Can't this be configured somehow?

**Pull request merged**